src/common_util.rs

```diff
@@ -50,14 +52,76 @@ pub fn strip_access_key(raw_menu_text: &str) -> String {
     result
 }
 
-/// A trait for implementing the boxed callback hack.
-pub(crate) trait IdleCallback: Send {
-    fn call(self: Box<Self>, a: &mut dyn WinHandler);
+pub(crate) type IdleCallback = Box<dyn for<'a> FnOnce(&'a mut dyn WinHandler) + Send>;
+
+/// A sharable queue. Similar to a `std::sync::mpsc` channel, this queue is implmented as two types:
+/// [SharedEnqueuer] and [SharedDequeuer]. The enqueuer can be cloned, while the dequeuer cannot.
+/// The major difference between this and a channel is that the queue can be checked for emptiness.
```

Comment:
It isn't really "emptiness," right? For one thing, the channel and the flag aren't protected by a common lock so the empty flag won't always be up-to-date. But also, if someone just `try_dequeue`ed the last message then the queue will be empty but the empty flag won't say so.

Two possible suggestions:
1) rename `is_empty` to something like `needs_wake`, which I think is closer to the intended semantics, or
2) nudge the user towards the correct usage by deleting `is_empty` and replacing `enqueue` with
```
#[must_use]
fn enqueue(&self, t: T) -> NeedsWake { ... }

enum NeedsWake {
  Yes,
  No,
}
```

Reply:
That API is closer to what I started out with, but I couldn't figure out an apt name for the return type of `enqueue` 😅 thanks for reading my mind on this one! I'll get this swapped out with a single `enqueue` method.